lib/screens/wallet/interactor/viewmodels/wallet_bloc.dart

```diff
@@ -1,20 +1,35 @@
 import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+import 'package:seeds/datasource/local/settings_storage.dart';
+import 'package:seeds/datasource/remote/model/profile_model.dart';
 import 'package:seeds/domain-shared/page_state.dart';
 import 'package:seeds/screens/wallet/interactor/mappers/user_account_state_mapper.dart';
 import 'package:seeds/screens/wallet/interactor/usecases/get_user_account.dart';
-import 'package:seeds/screens/wallet/interactor/viewmodels/wallet_event.dart';
-import 'package:seeds/screens/wallet/interactor/viewmodels/wallet_state.dart';
+
+part 'wallet_event.dart';
+
+part 'wallet_state.dart';
 
 /// --- BLOC
 class WalletBloc extends Bloc<WalletEvent, WalletState> {
-  WalletBloc() : super(WalletState.initial());
+  WalletBloc() : super(WalletState.initial()) {
+    on<OnLoadWalletData>(_onLoadWalletData);
+  }
 
-  @override
-  Stream<WalletState> mapEventToState(WalletEvent event) async* {
-    if (event is OnLoadWalletData) {
-      yield state.copyWith(pageState: PageState.loading);
-      final results = await GetUserAccountUseCase().run();
-      yield UserAccountStateMapper().mapResultToState(state, results);
+  Future<void> _onLoadWalletData(OnLoadWalletData event, Emitter<WalletState> emit) async {
+    emit(state.copyWith(pageState: PageState.loading));
+    final results = await GetUserAccountUseCase().run();
+    WalletState newState;
+    emit(newState = UserAccountStateMapper().mapResultToState(state, results));
+    if (newState.profile.status == ProfileStatus.citizen) {
```

Comment:
can this logic be moved to a use case?

Reply:
newState.profile.status is an enum, we can use a switch case instead of multiple ifs